File: packages_mjs/app-yaml-config/src/domain.ts
/**
 * Data models for AppYamlConfig.
 */

import { AppYamlConfig } from './core';

export interface LoadResult {
    filesLoaded: string[];
    errors: Array<{ file: string; error: string }>;
    appEnv: string | null;
    mergeOrder: string[];
}

export type ComputedDefinition = (config: AppYamlConfig) => any;
export type ComputedRegistry = Record<string, ComputedDefinition>;

export interface InitOptions {
    files: string[];
    configDir?: string;
    appEnv?: string;
    computedDefinitions?: ComputedRegistry;
}

export interface ResolutionSource {
    /** Source type: yaml, env */
    source: 'yaml' | 'env';
    /** Environment variable name that provided the value (if applicable) */
    envVar: string | null;
}

export interface BaseResolveOptions {
    applyEnvOverwrites?: boolean;
    removeMetaKeys?: boolean;
}

export interface BaseResult {
    name: string;
    config: Record<string, any>;
    envOverwrites: string[];
    resolutionSources: Record<string, ResolutionSource>;
}

File: packages_py/app_yaml_config/src/app_yaml_config/domain.py
"""Data models for AppYamlConfig."""

from dataclasses import dataclass, field
from typing import List, Dict, Any, Optional, Callable

@dataclass
class LoadResult:
    """Result of loading configuration files."""
    files_loaded: List[str] = field(default_factory=list)
    errors: List[Dict[str, Any]] = field(default_factory=list)
    app_env: Optional[str] = None
    merge_order: List[str] = field(default_factory=list)

# Type alias for computed functions: (config: AppYamlConfig) -> Any
ComputedDefinition = Callable[['AppYamlConfig'], Any]

@dataclass
class ResolutionSource:
    source: str  # 'yaml' | 'overwrite' | 'fallback'
    env_var: Optional[str] = None

@dataclass
class BaseResolveOptions:
    apply_env_overwrites: bool = True
    apply_fallbacks: bool = True
    remove_meta_keys: bool = True

@dataclass
class BaseResult:
    name: str
    config: Dict[str, Any]
    env_overwrites: List[str]
    resolution_sources: Dict[str, ResolutionSource]
